### ISSUE TITLE
Cannot run quickstart

### ISSUE THREAD

**Issue by dariober** [oclif/oclif#1211]
This a duplicate of issue #937 which has been closed but I'm having the same problem and a user in that thread reported the same very recently.

I'm following the quickstart steps but I don't get `./bin/dev`. As suggested in #937 I've done this first even if it is not in the quickstart instructions:

```
npm install oclif@latest -g
npm WARN deprecated debuglog@1.0.1: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm WARN deprecated readdir-scoped-modules@1.1.0: This functionality has been moved to @npmcli/fs
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.

added 770 packages in 45s

109 packages are looking for funding
  run `npm fund` for details
```

Then as per quickstart:

```
npx oclif generate mynewcli

Time to build an oclif CLI! Version: 4.0.4
? Select a module type CommonJS
Cloning into '/home/dario/Apollo/Apollo3/packages/apollo-cli/mynewcli'...
? npm package name mynewcli
? command bin name the CLI will export mynewcli
? description oclif example Hello World CLI
? author dariober @dariober
? license MIT
? Who is the GitHub owner of repository (https://github.com/OWNER/repo) dariober
? What is the GitHub name of repository (https://github.com/owner/REPO) mynewcli
? Select a package manager yarn
    force mynewcli/package.json
    force mynewcli/.gitignore
    force mynewcli/LICENSE

Changes to package.json were detected.

Running yarn install for you to install the required dependencies.
➤ YN0070: Migrating from Yarn 1; automatically enabling the compatibility node-modules linker 👍

➤ YN0000: ┌ Resolution step
➤ YN0061: │ querystring@npm:0.2.0 is deprecated: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
➤ YN0061: │ @oclif/color@npm:1.0.10 is deprecated: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
➤ YN0061: │ readdir-scoped-modules@npm:1.1.0 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0032: │ fsevents@npm:2.3.2: Implicit dependencies on node-gyp are discouraged
➤ YN0061: │ sinon@npm:16.0.0 is deprecated: 16.1.1
➤ YN0061: │ @npmcli/move-file@npm:1.1.2 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ @npmcli/move-file@npm:2.0.1 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ debuglog@npm:1.0.1 is deprecated: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
➤ YN0002: │ @oclif/core@npm:2.15.0 doesn't provide @types/node (pd4647), requested by ts-node
➤ YN0002: │ @oclif/core@npm:2.15.0 doesn't provide typescript (p47f40), requested by ts-node
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (p649ce), requested by @typescript-eslint/eslint-plugin
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (p8401f), requested by @typescript-eslint/parser
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (p61f6b), requested by eslint-config-xo-space
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (p8998f), requested by eslint-import-resolver-typescript
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (pf24c0), requested by eslint-plugin-import
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (p20c42), requested by eslint-plugin-mocha
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (p05c84), requested by eslint-plugin-node
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.15 doesn't provide eslint (pd71d6), requested by eslint-plugin-perfectionist
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (p03fc2), requested by eslint-config-xo-space
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (p3aaae), requested by eslint-plugin-mocha
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (p0cd6e), requested by eslint-plugin-node
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (pcfb1f), requested by eslint-plugin-unicorn
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 21s
➤ YN0000: ┌ Post-resolution validation
➤ YN0068: │ @jbrowse/core ➤ peerDependencies ➤ @emotion/react: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ @jbrowse/core ➤ peerDependencies ➤ @emotion/styled: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ chalk ➤ dependencies ➤ #ansi-styles: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ chalk ➤ dependencies ➤ #supports-color: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ cypress-mongodb ➤ dependencies ➤ bson: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ material-ui-popup-state ➤ peerDependencies ➤ @emotion/react: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ material-ui-popup-state ➤ peerDependencies ➤ @emotion/styled: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ material-ui-popup-state ➤ peerDependencies ➤ react-dom: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ mobx-react ➤ peerDependencies ➤ react-dom: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ mongoose-id-validator ➤ peerDependencies ➤ mongoose: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ yarn@npm:1.22.19 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yeoman-environment@npm:3.15.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yeoman-generator@npm:5.8.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yn@npm:3.1.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yocto-queue@npm:0.1.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 2s 587ms
➤ YN0000: ┌ Link step
➤ YN0007: │ yarn@npm:1.22.19 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 16s 410ms
➤ YN0000: Done with warnings in 40s 249ms
 ›   Warning: oclif update available from 3.17.2 to 4.0.4.
replacing <!-- usage --> in README.md
replacing <!-- commands --> in README.md
replacing <!-- toc --> in README.md

Created mynewcli in /home/dario/Apollo/Apollo3/packages/apollo-cli/mynewcli
```

After this I don't get `bin/dev` or `bin/run`:

```
ls -1 mynewcli/bin/
dev.cmd
dev.js
run.cmd
run.js
```

This is with node v18.17.1 on Ubuntu 20.

**Comment by dariober** [comment]
UPDATE: Selecting module type **ESM** instead of **CommonJS** at least gives something runnable with `./bin/dev.js` (ESM is *not* the default option). 

```
npx oclif generate mynewcli
Time to build an oclif CLI! Version: 4.0.4
? Select a module type ESM
Cloning into '/home/dario/Apollo/Apollo3/packages/apollo-cli/mynewcli'...
? npm package name mynewcli
? command bin name the CLI will export mynewcli
? description oclif example Hello World CLI (ESM)
? author dariober @dariober
? license MIT
? Who is the GitHub owner of repository (https://github.com/OWNER/repo) dariober
? What is the GitHub name of repository (https://github.com/owner/REPO) mynewcli
? Select a package manager yarn
    force mynewcli/package.json
    force mynewcli/.gitignore
    force mynewcli/LICENSE

Changes to package.json were detected.

Running yarn install for you to install the required dependencies.
➤ YN0070: Migrating from Yarn 1; automatically enabling the compatibility node-modules linker 👍

➤ YN0000: ┌ Resolution step
➤ YN0032: │ fsevents@npm:2.3.2: Implicit dependencies on node-gyp are discouraged
➤ YN0061: │ querystring@npm:0.2.0 is deprecated: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
➤ YN0061: │ @oclif/color@npm:1.0.10 is deprecated: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
➤ YN0061: │ sinon@npm:16.0.0 is deprecated: 16.1.1
➤ YN0061: │ @npmcli/move-file@npm:1.1.2 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ @npmcli/move-file@npm:2.0.1 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ readdir-scoped-modules@npm:1.1.0 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ debuglog@npm:1.0.1 is deprecated: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
➤ YN0002: │ @oclif/core@npm:2.15.0 doesn't provide @types/node (pd4647), requested by ts-node
➤ YN0002: │ @oclif/core@npm:2.15.0 doesn't provide typescript (p47f40), requested by ts-node
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (p3943b), requested by @typescript-eslint/eslint-plugin
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (p1ffab), requested by @typescript-eslint/parser
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (p83e56), requested by eslint-config-xo-space
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (p12aef), requested by eslint-import-resolver-typescript
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (pcd9d2), requested by eslint-plugin-import
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (pa9b61), requested by eslint-plugin-mocha
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (p8fbb3), requested by eslint-plugin-node
➤ YN0002: │ eslint-config-oclif-typescript@npm:3.0.13 doesn't provide eslint (p4620d), requested by eslint-plugin-perfectionist
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (p65213), requested by eslint-config-xo-space
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (pdc58c), requested by eslint-plugin-mocha
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (p48b6e), requested by eslint-plugin-node
➤ YN0002: │ eslint-config-oclif@npm:5.0.0 doesn't provide eslint (pcfb1f), requested by eslint-plugin-unicorn
➤ YN0002: │ oclif@npm:4.0.3 doesn't provide eslint (p79302), requested by eslint-plugin-perfectionist
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 20s 46ms
➤ YN0000: ┌ Post-resolution validation
➤ YN0068: │ @jbrowse/core ➤ peerDependencies ➤ @emotion/react: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ @jbrowse/core ➤ peerDependencies ➤ @emotion/styled: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ chalk ➤ dependencies ➤ #ansi-styles: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ chalk ➤ dependencies ➤ #supports-color: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ cypress-mongodb ➤ dependencies ➤ bson: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ material-ui-popup-state ➤ peerDependencies ➤ @emotion/react: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ material-ui-popup-state ➤ peerDependencies ➤ @emotion/styled: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ material-ui-popup-state ➤ peerDependencies ➤ react-dom: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ mobx-react ➤ peerDependencies ➤ react-dom: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0068: │ mongoose-id-validator ➤ peerDependencies ➤ mongoose: No matching package in the dependency tree; you may not need this rule anymore.
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ yarn@npm:1.22.19 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yeoman-environment@npm:3.15.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yeoman-generator@npm:5.8.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yn@npm:3.1.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yocto-queue@npm:0.1.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 2s 397ms
➤ YN0000: ┌ Link step
➤ YN0007: │ yarn@npm:1.22.19 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 13s 262ms
➤ YN0000: Done with warnings in 35s 957ms
 ›   Warning: oclif update available from 4.0.3 to 4.0.4.
replacing <!-- usage --> in README.md
replacing <!-- commands --> in README.md
replacing <!-- toc --> in README.md

Created mynewcli in /home/dario/Apollo/Apollo3/packages/apollo-cli/mynewcli
```

```
npx oclif generate mynewcli
```

```
./bin/dev.js hello world
hello world! (./src/commands/hello/world.ts)
```

**Comment by dariober** [comment]
I see the quickstart page has changed to `./bin/dev.js hello world`. However, this still doesn't work unless you select module type ESM:

```
./bin/dev.js hello world
/usr/bin/env: ‘ts-node’: No such file or directory
```
